src/Microsoft.AspNetCore.ResponseCompression/ResponseCompressionProvider.cs

```diff
@@ -0,0 +1,102 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using Microsoft.AspNetCore.Http;
+using Microsoft.Extensions.Options;
+using Microsoft.Extensions.Primitives;
+using Microsoft.Net.Http.Headers;
+
+namespace Microsoft.AspNetCore.ResponseCompression
+{
+    /// <inheritdoc />
+    public class ResponseCompressionProvider : IResponseCompressionProvider
+    {
+        private readonly ICompressionProvider[] _providers;
+        private readonly HashSet<string> _mimeTypes;
+
+        /// <summary>
+        /// If no compression providers are specified then GZip is used by default.
+        /// </summary>
+        /// <param name="providers">Compression providers to use, if any.</param>
+        /// <param name="options"></param>
+        public ResponseCompressionProvider(IEnumerable<ICompressionProvider> providers, IOptions<ResponseCompressionOptions> options)
+        {
+            _providers = providers.ToArray();
+            if (_providers.Length == 0)
+            {
+                _providers = new [] { new GzipCompressionProvider() };
```

Comment:
Is it not better to call `TryAddTransient<ICompressionProvider, GzipCompressionProvider>()` (or `TryAddSingleton`) in `AddResponseCompression`?